tds_test.go

```diff
@@ -694,6 +694,52 @@ func TestBadCredentials(t *testing.T) {
 	_ = testConnectionBad(t, params.URL().String())
 }
 
+func TestLeakedConnections(t *testing.T) {
+	goodParams := testConnParams(t)
+	badParams := testConnParams(t)
+	badParams.Database = "unknown_db"
+
+	// Connecting with good credentials should not fail
+	goodConn, err := sql.Open("mssql", goodParams.URL().String())
+	if err != nil {
+		t.Fatal("Open connection failed:", err.Error())
+	}
+	err = goodConn.Ping()
+	if err != nil {
+		t.Fatal("Ping with good credentials should not fail, but got error:", err.Error())
+	}
+	// Remember the number of open connections, excluding the current one
+	var openConnections int
+	err = goodConn.QueryRow("SELECT COUNT(*) AS openConnections FROM sys.dm_exec_connections WHERE session_id!=@@SPID").Scan(&openConnections)
```

Comment:
some folks have to run the tests against shared servers. Can you capture the `local_net_address` value from the `goodConn`'s SPID and limit the search to rows that match it?

Reply:
sure, good point